### ISSUE TITLE
Tensor vs. vmap computations

### ISSUE THREAD

**Issue by EtienneCmb** [brainets/hoi#5]
# Goal

Comparison between the loop-free tensor implementation of the entropy vs. using `jax.vmap` looping over the first axis. 

# Comparison

- Which implementation is the fastest?
- Which one takes less memory (GPU)?

**Comment by EtienneCmb** [comment]
Vmap scales better with larger data

![image](https://github.com/brainets/hoi/assets/15892073/dffe27cc-1ad5-4ab2-811c-e9ad22867d2a)

**Comment by EtienneCmb** [comment]
Same on cpu
![image](https://github.com/brainets/hoi/assets/15892073/5e6fa1cf-1b13-4ad5-8215-b0dbd2835333)
